File: golang/requester/request-response.go
package requester

import (
	"context"
	"log"

	"github.com/rsocket/rsocket-go/payload"
)

func ExecRequestResponse() {
	log.Println("====ExecRequestResponse====")

	cli, err := BuildClient()
	defer cli.Close()
	// Send request
	result, err := cli.RequestResponse(payload.NewString("你好", "世界")).Block(context.Background())
	if err != nil {
		panic(err)
	}
	log.Println("[Request-Response] response:", result)
	log.Println("[Request-Response] data:", result.DataUTF8())
}

File: rust/src/requester/request_response.rs
use futures::prelude::*;
use rsocket_rust::prelude::*;

use crate::requester::client;

pub fn exec_request_response() {
    println!("====ExecRequestResponse====");
    let cli = client::build_client();

    for n in 0..3 {
        let pa = Payload::builder()
            .set_data_utf8("Hello World!")
            .set_metadata_utf8(&format!("#{}", n))
            .build();
        let resp = cli.request_response(pa).wait().unwrap();
        println!("<<<<<<<< : {:?}", resp.data());
    }

    let _ = cli.on_close();
}

